src/lib/wallet/MultipleHttpProvider.js

```diff
@@ -33,8 +33,10 @@ export class MultipleHttpProvider extends HttpProvider {
   send(payload, callback) {
     const { endpoints, strategy, retries, loggedProviders } = this
 
+    const filteredEndpoints = endpoints.filter(({ provider }) => !loggedProviders.has(provider))
```

Comment:
this should be reverted, loggedProviders is only for logging rule (1 per session per url)

Reply:
1. remove too many req from regex, elude it from logging logic 
2. add separate check for rate limit error . it should be processed via separate flow, not with logging rules
3. when it happens, you need to move/keep rpc failing with this error for X seconds (60 sec ?) and exclude from shuffle. but after this time period ends - you bring them back. you maybe do not need instance var, just

```javascript
endpoints.splice(endpoints.findIndex(failingUrl), 1)
setTimeout(() => endpoints.push(failingUrl))
```